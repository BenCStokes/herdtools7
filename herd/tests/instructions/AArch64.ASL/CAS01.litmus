AArch64 CAS01
(* CAS, ok case *)

{
int64_t x = 3;
int64_t 0: X1 = 3;
int64_t 0: X2 = 4;
        0: X3 = x;
}

P0 ;
CAS X1, X2, [X3] ;

exists (0: X1 = 3 /\ x = 4)
