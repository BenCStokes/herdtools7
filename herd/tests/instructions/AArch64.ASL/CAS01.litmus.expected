Test CAS01 Allowed
States 1
0:X1=3; [x]=4;
Ok
Witnesses
Positive: 1 Negative: 0
Condition exists (0:X1=3 /\ [x]=4)
Observation CAS01 Always 1 0
Hash=b8931ab63fae4945344e617434d5bfd5

